### ISSUE TITLE
ChaiScript receives in invalid return value from C++ method.

### ISSUE THREAD

**Issue by shockdot** [ChaiScript/ChaiScript#385]
* Compiler Used: VS 2017
 * Operating System: Windows 7
 * Architecture (ARM/x86/32bit/64bit/etc):  x64
 

### Expected Behavior 
I am trying to have a ChaiScript execute a C++ function and compare the value against another value. In my example, the value 5 should print to the console, then True, then a string saying "Condition met.".

### Actual Behavior 
The value returned by the C++ function seems to not be the real return value... Instead it returns -858993460, and thus causes the comparison to fail....

### Minimal Example to Reproduce Behavior
**ChaiScript.h**

```
#pragma once

// VS2017 Bug, when this gets fixed remove this
#pragma warning( disable : 4710 )

#pragma warning( push )
#pragma warning( disable : 4061 )
#pragma warning( disable : 4068 )
#pragma warning( disable : 4191 )
#pragma warning( disable : 4355 )
#pragma warning( disable : 4365 )
#pragma warning( disable : 4371 )
#pragma warning( disable : 4464 )
#pragma warning( disable : 4514 )
#pragma warning( disable : 4571 )
#pragma warning( disable : 4623 )
#pragma warning( disable : 4625 )
#pragma warning( disable : 4626 )
#pragma warning( disable : 4668 )
#pragma warning( disable : 4710 )
#pragma warning( disable : 4774 )
#pragma warning( disable : 4820 )
#pragma warning( disable : 5026 )
#pragma warning( disable : 5027 )
#include <chaiscript\chaiscript.hpp>

class ChaiScript
{
public:
   ChaiScript(
      const std::string& script_name);

   ~ChaiScript();

   const int GetValueForID(
      const int id) const;

   chaiscript::ChaiScript chai;

private:
};
#pragma warning( pop )

```

**ChaiScript.cpp**

```
#include "ChaiScript.h"

ChaiScript::ChaiScript(
   const std::string& script_name)
{
   chai.eval_file(script_name);
   chai.add(chaiscript::fun(&ChaiScript::GetValueForID, this), "GetValueForID");
}

ChaiScript::~ChaiScript()
{

}

const int ChaiScript::GetValueForID(
   const int id) const
{
   (void)id;
   return 5;
}
```

**Main Func**
```
#include "stdafx.h"
#include "ChaiScript.h"

int main()
{
   ChaiScript* script = new ChaiScript("ChaiSample.chai");
   script->chai.eval<std::function<void(int)>>("CallMe")(1);

   getchar();
}
```

**Sample.chai**
```
def CallMe(id)
{
	var ReceivedValue = GetValueForID(id);
	print(ReceivedValue);
	print(ReceivedValue >= 4);
	if (ReceivedValue >= 4)
	{
		print("Condition met.");
	}
}
```

**Comment by shockdot** [comment]
Strange, removing the const return value fixed it.... Is there a reason behind this?

**Comment by lefticus** [comment]
This was fixed in #340 which is on develop. But also, you should almost never return value type that is `const` qualified, it probably doesn't do what you think it would do.
